src/core/project/qgsprojectviewsettings.cpp

```diff
@@ -71,6 +74,17 @@ void QgsProjectViewSettings::setPresetFullExtent( const QgsReferencedRectangle &
   emit presetFullExtentChanged();
 }
 
+void QgsProjectViewSettings::setMainCanvasOpensAtProjectExtent( bool state )
```

Comment:
The parameter name state should be renamed to something more appropriate (as well as function name, I commented in the .h already).